### PR TITLE
pod: Add secret subpath

### PR DESCRIPTION
**Pull request by ravanelli**
- Add option for mounting secrets using subpath in /etc;
 - Change the mountpath if it is a brew-ca certificate

### CODE REVIEW

**Comment by ravanelli** [comment]
It is required for https://github.com/coreos/fedora-coreos-pipeline/pull/836

File: vars/pod.groovy [hunk 1]
```diff
@@ -80,11 +80,18 @@ def call(params = [:], Closure body) {
     }
 
     params['secrets'].eachWithIndex { secret, i ->
-        podObj['spec']['volumes'] += ['name': "secret-${i}".toString(), 'secret': [secretName: secret]]
-        podObj['spec']['containers'][0]['volumeMounts'] += ['name': "secret-${i}".toString(), 'mountPath': "/run/kubernetes/secrets/${secret}".toString()]
-
+        if (secret.contains(":")) {
+            def key
+            (secret, key, mountPath) = secret.split(":")
+            podObj['spec']['volumes'] += ['name': secret, 'secret': [secretName: secret, items: [['key':key, 'path': key]]]]
```
Comment (jlebon) [inline vars/pod.groovy:86]:
```suggestion
            podObj['spec']['volumes'] += ['name': secret, 'secret': [secretName: secret, items: [['key': key, 'path': key]]]]
```

Comment (jlebon) [reply vars/pod.groovy:86]: Multiple keys from the same secret may be used multiple times. So for the secret name, maybe it's best to keep it `secret-${i}`?

**Review by jlebon — APPROVED** [review]
LGTM!

**Pull request merged**